authors/apps/articles/views.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+from rest_framework import generics
+from .models import Article
+from .serializers import CreateArticleAPIViewSerializer, UpdateArticleAPIVIEWSerializer
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from .renderers import ArticleJSONRenderer
+from django.template.defaultfilters import slugify
+from ..authentication.backends import JWTAuthentication
+from rest_framework.response import Response
+from rest_framework import status
+from ..authentication.models import User
+from django.shortcuts import get_object_or_404
+
+
+class ListCreateArticleAPIView(generics.ListCreateAPIView):
+
+
+    renderer_classes = (ArticleJSONRenderer, )
+    queryset = Article.objects.all()
+    serializer_class = CreateArticleAPIViewSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def create(self, request):
+        """
+            This method creates user articles
+        """
+        article = request.data.get('article', {})
+        user_info = JWTAuthentication().authenticate(request)
+        article['author'] = user_info[0]
+
+        slug = slugify(article["title"]).replace("_", "-")
+        slug = slug + "-" + str(uuid.uuid4()).split("-")[-1]
+        article["slug"] = slug
+
+        serializer = self.serializer_class(data=article)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(author=user_info[0])
+        data = serializer.data
+        data["message"] = "Your Article has been created successfully"
```

Comment:
This is a no big deal, but massaging of the data to be returned could be done in the `Renderer classes` since you already have one.

Reply:
Okay